zallet/src/components/wallet/connection.rs

```diff
@@ -37,6 +38,10 @@ pub(super) type WalletPool = deadpool::managed::Pool<WalletManager>;
 
 pub(crate) struct WalletManager {
     inner: deadpool_sqlite::Manager,
+    /// Connection pools are thread-safe, but SQLite does not reliably follow the busy
+    /// handler (configured by `rusqlite` to a timeout after 5s), so we explicitly guard
+    /// against SQLite `DatabaseBusy` errors.
```

Comment:
What does "not reliably follow the busy handler" mean? (I understand that using this guard will avoid `DatabaseBusy` errors; I just don't understand the wording about what will happen if we don't.)

Reply:
https://www.sqlite.org/c3ref/busy_handler.html

> The presence of a busy handler does not guarantee that it will be invoked when there is lock contention. If SQLite determines that invoking the busy handler could result in a deadlock, it will go ahead and return SQLITE_BUSY to the application instead of invoking the busy handler.